wasm-runtime/std/src/lib.rs

```diff
@@ -26,16 +26,22 @@ extern crate pwasm_alloc;
 
 #[lang = "panic_fmt"]
 #[no_mangle]
-pub fn panic_fmt() -> ! {
-	  loop {}
+pub extern fn panic_fmt(_fmt: ::core::fmt::Arguments, _file: &'static str, _line: u32, _col: u32) {
+	unsafe {
+		ext_print_utf8(_file.as_ptr() as *const u8, _file.len() as u32);
+		ext_print_num(_line as u64);
+		ext_print_num(_col as u64);
+		::core::intrinsics::abort()
```

Comment:
in a future PR we should be able to pass the reason for failure to the externalities.